### DISCUSSION TITLE
`started_at` and `completed_at` of each step is not accurate

### DISCUSSION THREAD

**Discussion by kota65535** [community/community#41351] [Apps, API and Webhooks]
### Select Topic Area

Bug

### Body

[Get a job for a workflow run](https://docs.github.com/en/rest/actions/workflow-jobs?apiVersion=2022-11-28#get-a-job-for-a-workflow-run) API returns start time and completion time of each step of the job, but they are not accurate.

To be more specific, decimal places of `started_at` and `completed_at` are truncated like this response below:

```
{
  "name": "Run terraform plan",
  "status": "completed",
  "conclusion": "success",
  "number": 6,
  "started_at": "2022-12-11T02:47:53.000+09:00",
  "completed_at": "2022-12-11T02:48:32.000+09:00"
}
```

When I investigate the self-hosted runner log, it appears that the start time and the finish time are sent with 7 decimal places.

```
[2022-12-10 17:48:32Z INFO ExecutionContext] Publish step telemetry for current step {
  "action": "sh",
  "type": "run",
  "stage": "Main",
  "stepId": "b7af8ad4-115d-5e0e-033e-01363c8ce2e3",
  "stepContextName": "__run_3",
  "result": "succeeded",
  "errorMessages": [],
  "executionTimeInSeconds": 40,
  "startTime": "2022-12-10T17:47:53.1412719Z",
  "finishTime": "2022-12-10T17:48:32.7506489Z"
}.
```

I think [Get a job for a workflow run](https://docs.github.com/en/rest/actions/workflow-jobs?apiVersion=2022-11-28#get-a-job-for-a-workflow-run) API should return this start time and finish time as is. Is there any reason not to do so?

**Comment by dlvhdr** [comment]
I would also like this supported. 
Currently github doesn't have an api to get a specific step logs, this would at least allow us to correlate the logs with the timestamps.

**Comment by maschwenk** [comment]
Not having this makes it very hard for agents like Claude Code to read Github Job logs with any kind of context-efficiency
